lib/grammers-mtproto/src/transport/mod.rs

```diff
@@ -91,8 +93,17 @@ pub trait Transport {
     fn pack(&mut self, buffer: &mut DequeBuffer<u8>);
 
     /// Unpacks the input buffer in-place.
-    fn unpack(&mut self, buffer: &[u8]) -> Result<UnpackedOffset, Error>;
+    /// Subsequent calls to `unpack` should be made with the same buffer only
+    /// with the old data removed and new data appended.
```

Comment:
> with the old data removed and new data appended

This part seems misleading. The old data will still be there until `unpack` returns an offset. Once the offset is returned, the processed data is removed.

Maybe this reads better, or maybe you can think of better wording:

```suggestion
    /// Subsequent calls to `unpack` should be made with the same buffer,
    /// with the data on the ranges from previous `UnpackedOffset` removed.
```

Reply:
Done.